### ISSUE TITLE
Sign-in with AAD Auth: AADSTS50000: There was an error issuing a token or an issue with our sign-in service.

### ISSUE THREAD

**Issue by lippertmarkus** [microsoft/navcontainerhelper#2390]
**Describe the issue**
I'm trying to create a BC container with AAD auth using the script below. The same script works fine in two different tenants, but in our company tenant it shows the *There was an error issuing a token or an issue with our sign-in service* during SSO to the BC container.

**Scripts used to create container and cause the issue**
```powershell
$appIdUriPrefix = "selfservice-ppitest"
$tenantName = "cosmoconsult.com"
$appIdUri = "https://$appIdUriPrefix.$tenantName"

Connect-AzureAD
$ids = Create-AadAppsForBC -appIdUri $appIdUri -publicWebBaseUrl "https://mybcvm.westeurope.cloudapp.azure.com/bcaad19" -useCurrentAzureAdConnection -SingleTenant -IncludeExcelAadApp -IncludePowerBiAadApp -IncludeEmailAadApp -IncludeApiAccess

New-BcContainer -accept_eula -accept_outdated -containerName bcaad19 `
    -artifactUrl (Get-BCArtifactUrl -version 19.5 -type Sandbox) -licenseFile "<mylicenseUrl>" `
    -useSSL -useTraefik -authenticationEMail "<myemail>" `
    -auth AAD -AadTenant $tenantName -AadAppId "$($ids.SsoAdAppId)" -AadAppIdUri $appIdUri
```

**Full output of scripts**
```
BcContainerHelper is version 3.0.4
BcContainerHelper is running as administrator
Host is Microsoft Windows Server 2019 Datacenter - ltsc2019
Docker Client Version is 20.10.14
Docker Server Version is 20.10.14
Downloading application artifact /sandbox/19.5.36567.38753/w1
Downloading C:\Users\vmadmin\AppData\Local\Temp\4\fd090985-fa83-480f-b38f-90b3f3d3b83d.zip
Unpacking application artifact to tmp folder using Expand-Archive
Downloading platform artifact /sandbox/19.5.36567.38753/platform
Downloading C:\Users\vmadmin\AppData\Local\Temp\4\110b46fd-6efe-4fab-9ada-75dd99f83ef0.zip
Unpacking platform artifact to tmp folder using Expand-Archive
Downloading Prerequisite Components
Downloading c:\bcartifacts.cache\sandbox\19.5.36567.38753\platform\Prerequisite Components\Open XML SDK 2.5 for Microsoft Office\OpenXMLSDKv25.msi
Downloading c:\bcartifacts.cache\sandbox\19.5.36567.38753\platform\Prerequisite Components\IIS URL Rewrite Module\rewrite_2.0_rtw_x64.msi
Downloading c:\bcartifacts.cache\sandbox\19.5.36567.38753\platform\Prerequisite Components\DotNetCore\DotNetCore.1.0.4_1.1.1-WindowsHosting.exe
Fetching all docker images
Fetching all docker volumes
Enabling SSL as otherwise all clients will see mixed HTTP / HTTPS request, which will cause problems e.g. on the mobile and modern windows clients
Using image mcr.microsoft.com/businesscentral:10.0.17763.2686
PublicDnsName is mybcvm.westeurope.cloudapp.azure.com
Creating Container bcaad19
Style: sandbox
Multitenant: Yes
Version: 19.5.36567.38753
Platform: 19.0.36528.38726
Generic Tag: 1.0.2.1
Container OS Version: 10.0.17763.2686 (ltsc2019)
Host OS Version: 10.0.17763.2686 (ltsc2019)
Using process isolation
Using locale en-US
Adding special CheckHealth.ps1 to enable Traefik support
Disabling the standard eventlog dump to container log every 2 seconds (use -dumpEventLog to enable)
Using license file https://ccppi.blob.core.windows.net/lic/devlic-bc19-de
Downloading C:\ProgramData\BcContainerHelper\Extensions\bcaad19\my\license
Additional Parameters:
-e webserverinstance=bcaad19
-e publicdnsname=mybcvm.westeurope.cloudapp.azure.com
-l "traefik.protocol=https"
-l "traefik.web.frontend.rule=PathPrefix:/bcaad19"
-l "traefik.web.port=443"
-l "traefik.soap.frontend.rule=PathPrefix:/bcaad19soap;ReplacePathRegex: ^/bcaad19soap(.*) /BC$1"
-l "traefik.soap.port=7047"
-l "traefik.rest.frontend.rule=PathPrefix:/bcaad19rest;ReplacePathRegex: ^/bcaad19rest(.*) /BC$1"
-l "traefik.rest.port=7048"
-l "traefik.dev.frontend.rule=PathPrefix:/bcaad19dev;ReplacePathRegex: ^/bcaad19dev(.*) /BC$1"
-l "traefik.dev.port=7049"
-l "traefik.snap.frontend.rule=PathPrefix:/bcaad19snap;ReplacePathRegex: ^/bcaad19snap(.*) /BC$1"
-l "traefik.snap.port=7083"
-l "traefik.dl.frontend.rule=PathPrefixStrip:/bcaad19dl"
-l "traefik.dl.port=8080"
-l "traefik.dl.protocol=http"
-l "traefik.enable=true"
-l "traefik.frontend.entryPoints=https"
--env customNavSettings=ValidAudiences=9ffc2893-7dea-4c10-8f3f-403a608af0ec;https://api.businesscentral.dynamics.com,DisableTokenSigningCertificateValidation=True,ExtendedSecurityTokenLifetime=24,ClientServicesCredentialType=NavUserPassword,PublicODataBaseUrl=https://mybcvm.westeurope.cloudapp.azure.com/bcaad19rest/odata,PublicSOAPBaseUrl=https://mybcvm.westeurope.cloudapp.azure.com/bcaad19soap/ws,PublicWebBaseUrl=https://mybcvm.westeurope.cloudapp.azure.com/bcaad19
Files in C:\ProgramData\BcContainerHelper\Extensions\bcaad19\my:
- AdditionalOutput.ps1
- CheckHealth.ps1
- license
- MainLoop.ps1
- SetupVariables.ps1
- updatecontainerhosts.ps1
Creating container bcaad19 from image mcr.microsoft.com/businesscentral:10.0.17763.2686
de05a6b257c03fd2ee214b76a060301eb6c508d8efc0fc82c3deb4df41276887
Waiting for container bcaad19 to be ready
Using artifactUrl https://bcartifacts.azureedge.net/sandbox/19.5.36567.38753/w1
Using installer from C:\Run\150-new
Installing Business Central
Installing from artifacts
Starting Local SQL Server
Starting Internet Information Server
Copying Service Tier Files
c:\dl\sandbox\19.5.36567.38753\platform\ServiceTier\Program Files
c:\dl\sandbox\19.5.36567.38753\platform\ServiceTier\System64Folder
Copying PowerShell Scripts
c:\dl\sandbox\19.5.36567.38753\platform\WindowsPowerShellScripts\Cloud\NAVAdministration
c:\dl\sandbox\19.5.36567.38753\platform\WindowsPowerShellScripts\WebSearch
Copying dependencies
Copying ReportBuilder
Importing PowerShell Modules
Determining Database Collation from c:\dl\sandbox\19.5.36567.38753\w1\BusinessCentral-W1.bak
Restoring CRONUS Demo Database
Setting CompatibilityLevel for tenant on localhost\SQLEXPRESS
Exporting Application to CRONUS
Removing Application from tenant
Modifying Business Central Service Tier Config File for Docker
Creating Business Central Service Tier
Installing SIP crypto provider: 'C:\Windows\System32\NavSip.dll'
Copying Web Client Files
c:\dl\sandbox\19.5.36567.38753\platform\WebClient\Microsoft Dynamics NAV
Copying Client Files
c:\dl\sandbox\19.5.36567.38753\platform\LegacyDlls\program files\Microsoft Dynamics NAV
c:\dl\sandbox\19.5.36567.38753\platform\LegacyDlls\program files\Microsoft Dynamics NAV
c:\dl\sandbox\19.5.36567.38753\platform\LegacyDlls\systemFolder
Copying ModernDev Files
c:\dl\sandbox\19.5.36567.38753\platform
c:\dl\sandbox\19.5.36567.38753\platform\ModernDev\program files\Microsoft Dynamics NAV
Copying additional files
Copying ConfigurationPackages
C:\dl\sandbox\19.5.36567.38753\platform\ConfigurationPackages
Copying Test Assemblies
C:\dl\sandbox\19.5.36567.38753\platform\Test Assemblies
Copying Extensions
C:\dl\sandbox\19.5.36567.38753\w1\Extensions
Copying Applications
C:\dl\sandbox\19.5.36567.38753\platform\Applications
Starting Business Central Service Tier
Importing license file
Copying Database on localhost\SQLEXPRESS from tenant to default
Taking database tenant offline
Copying database files
Attaching files as new Database default
Putting database tenant back online
Mounting tenant database
Mounting Database for default on server localhost\SQLEXPRESS with AllowAppDatabaseWrite = False
Sync'ing Tenant
Tenant is Operational
Stopping Business Central Service Tier
Installation took 205 seconds
Installation complete
Initializing...
Setting host.containerhelper.internal to 172.17.32.1 in container hosts file
Starting Container
Hostname is bcaad19
PublicDnsName is mybcvm.westeurope.cloudapp.azure.com
Using AccessControlService Authentication
Creating Self Signed Certificate
Self Signed Certificate Thumbprint 69E69B33BB8B4E756D631E53EE1796E23D1D04C8
DNS identity mybcvm.westeurope.cloudapp.azure.com
Modifying Service Tier Config File with Instance Specific Settings
Modifying Service Tier Config File with settings from environment variable
Setting ValidAudiences to 9ffc2893-7dea-4c10-8f3f-403a608af0ec;https://api.businesscentral.dynamics.com
Setting DisableTokenSigningCertificateValidation to True
Setting ExtendedSecurityTokenLifetime to 24
Setting ClientServicesCredentialType to NavUserPassword
Setting PublicODataBaseUrl to https://mybcvm.westeurope.cloudapp.azure.com/bcaad19rest/odata
Setting PublicSOAPBaseUrl to https://mybcvm.westeurope.cloudapp.azure.com/bcaad19soap/ws
Setting PublicWebBaseUrl to https://mybcvm.westeurope.cloudapp.azure.com/bcaad19
Starting Service Tier
Registering event sources
Creating DotNetCore Web Server Instance
Enabling Financials User Experience
Using license file 'c:\run\my\license'
Import License
Dismounting Tenant
Mounting Tenant
Setting AadTenantId to cosmoconsult.com
Mounting Database for default on server localhost\SQLEXPRESS with AllowAppDatabaseWrite = False
Sync'ing Tenant
Tenant is Operational
Creating http download site
Setting SA Password and enabling SA
Creating bcadmin as SQL User and add to sysadmin
Creating SUPER user
Container IP Address: 172.17.38.35
Container Hostname  : bcaad19
Container Dns Name  : mybcvm.westeurope.cloudapp.azure.com
Web Client          : https://mybcvm.westeurope.cloudapp.azure.com/bcaad19/?tenant=default
Dev. Server         : https://mybcvm.westeurope.cloudapp.azure.com
Dev. ServerInstance : BC
Dev. Server Tenant  : default
Setting bcaad19-default to 127.0.0.1 in container hosts file

Files:
http://mybcvm.westeurope.cloudapp.azure.com:8080/ALLanguage.vsix
http://mybcvm.westeurope.cloudapp.azure.com:8080/certificate.cer

Container Total Physical Memory is 16.0Gb
Container Free Physical Memory is 8.2Gb

Initialization took 70 seconds
Ready for connections!
Reading CustomSettings.config from bcaad19
Creating Desktop Shortcuts for bcaad19
Container bcaad19 successfully created
Because of Traefik, the following URLs need to be used when accessing the container from outside your Docker host:
Web Client:        https://mybcvm.westeurope.cloudapp.azure.com/bcaad19?tenant=default
SOAP WebServices:  https://mybcvm.westeurope.cloudapp.azure.com/bcaad19soap
OData WebServices: https://mybcvm.westeurope.cloudapp.azure.com/bcaad19rest
Dev Service:       https://mybcvm.westeurope.cloudapp.azure.com/bcaad19dev
Snapshot Service:  https://mybcvm.westeurope.cloudapp.azure.com/bcaad19snap
File downloads:    https://mybcvm.westeurope.cloudapp.azure.com/bcaad19dl

Use:
Get-BcContainerEventLog -containerName bcaad19 to retrieve a snapshot of the event log from the container
Get-BcContainerDebugInfo -containerName bcaad19 to get debug information about the container
Enter-BcContainer -containerName bcaad19 to open a PowerShell prompt inside the container
Remove-BcContainer -containerName bcaad19 to remove the container again
docker logs bcaad19 to retrieve information about URL's again
```

**Screenshots**
![image](https://user-images.githubusercontent.com/1463140/161992778-9b8f01e8-cc53-4e84-8558-d62d18e7f396.png)

**Additional context**

**Comment by freddydk** [comment]
Thanks for the update.
Are you logging in with @cosmoconsult accounts?
Have you tried to remove the -singletenant?

**Comment by lippertmarkus** [comment]
Just tried but still getting the error.

**Comment by lippertmarkus** [comment]
Also tried successfully in a completely fresh CDX environment. App Registrations are identical between the two tenants.
Any ideas on which AAD settings we could try to compare?

**Comment by freddydk** [comment]
Will try to repro this (Just came home from Directions NA)

**Comment by lippertmarkus** [comment]
We found it.
We had a policy where app registration permissions need consent when they are not in an allowed list.

So although the app registration showed that no admin consent is needed:
![MicrosoftTeams-image (13)](https://user-images.githubusercontent.com/1463140/162469463-838b5b42-9beb-4d4d-b2c8-9a6f0c058134.png)

It only worked after we granted admin consent on all permissions.

Thanks anyway and enjoy your weekend after your many Direction Sessions :)
